lib/StyleFieldPlugin.js

```diff
@@ -0,0 +1,95 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Burhanuddin Udaipurwala @burhanuday
+*/
+
+"use strict";
+
+const path = require("path");
+const DescriptionFileUtils = require("./DescriptionFileUtils");
+
+/** @typedef {import("./Resolver")} Resolver */
+/** @typedef {import("./Resolver").JsonObject} JsonObject */
+/** @typedef {import("./Resolver").ResolveRequest} ResolveRequest */
+/** @typedef {import("./Resolver").ResolveStepHook} ResolveStepHook */
+
+const alreadyTriedStyleField = Symbol("alreadyTriedStyleField");
+
+module.exports = class StyleFieldPlugin {
+	/**
+	 * @param {string | ResolveStepHook} source source
+	 * @param {string} name fieldName
+	 * @param {string | ResolveStepHook} target target
+	 */
+	constructor(source, name, target) {
+		this.source = source;
+		this.name = name;
+		this.target = target;
+	}
+
+	/**
+	 * @param {Resolver} resolver the resolver
+	 * @returns {void}
+	 */
+	apply(resolver) {
+		const target = resolver.ensureHook(this.target);
+		resolver
+			.getHook(this.source)
+			.tapAsync("StyleFieldPlugin", (request, resolveContext, callback) => {
+				// if request.context.issuer is not a css file, don't resolve to style field
+				const isIssuerAStyleFile =
+					request.context &&
+					// @ts-ignore
+					request.context.issuer &&
+					// @ts-ignore
+					request.context.issuer.endsWith(".css");
```

Comment:
todo: check for all known css extensions
todo2: allow consumer to configure extensions array